modules/templates/template_proxy_request.rb

```diff
@@ -0,0 +1,35 @@
+require "proxy/util"
+require 'proxy/request'
+require 'uri'
+
+module Proxy::Templates
+
+  class TemplateProxyRequest < ::Proxy::HttpRequest::ForemanRequest
+    include Proxy::Log
+
+    def get_template(kind, request, params)
+      url = Proxy::Templates::Plugin.settings.template_url
+      proxy_ip = URI.parse(url).host
+      opts = params.clone.merge(:url => url)
+      opts.delete("kind")
+      opts.delete("splat")
+      opts.delete("captures")
+      proxy_headers = Hash[request.env.select{|k,v| k =~ /^HTTP_/ && k !~ /^HTTP_(VERSION|HOST)$/}.map{|k,v| [k[5..-1],v]}].merge("X-Forwarded-For" => "#{request.ip}, #{proxy_ip}")
```

Comment:
Didn't realise ruby does a splat under the hood when assigning an array to multiple vars, so this works under 1.8.7. None of the tests cover the filter bit though, which would be useful. Probably worth extracting this line into a method too.